data_tools/compute_coding_progress.py

```diff
@@ -6,36 +6,37 @@
 
 if (len(sys.argv) != 2):
     print ("Usage python compute_coding_progress.py coda_crypto_token")
-    exit(1)
+        exit(1)
 
 CODA_CRYPTO_TOKEN_PATH = sys.argv[1]
-
 fcw.init_client(CODA_CRYPTO_TOKEN_PATH)
-data = {}
-
-ids = fcw.get_dataset_ids()
-
-data['coding_progress'] = {}
 
-for id in ids:
-    data['coding_progress'][id] = {}
-    messages = []
-    messages_with_labels = 0
-
-    # New scheme
-    metrics = fcw.get_dataset_metrics(id)
-    if metrics != None:
-        data['coding_progress'][id] = metrics   
-        continue
-        
+def compute_coding_progress(ids,messages,metrics):
+    data = {}
+    data['coding_progress'] = {}
+    for id in ids:
+        data['coding_progress'][id] = {}  
+        messages_with_labels = 0
+        # New scheme
+        if metrics != None:
+            data['coding_progress'][id] = metrics   
+            continue 
+            if len(message["Labels"]) > 0:
+                messages_with_labels += 1
+        data['coding_progress'][id]['messages_count'] = len(messages)
+        data['coding_progress'][id]['messages_with_label'] = messages_with_labels
+    data["last_update"] = strftime("%Y-%m-%d %H:%M:%S", gmtime())
+    return json.dumps(data, indent=2)
```

Comment:
This returns a string, it's better to have the code that computes the answer separate from the code that does something with that answer. 

So in this case, I think we should be returning data, and then the place that calls this function calling dumps